### ISSUE TITLE
SHT3X-D wemos shield not working

### ISSUE THREAD

**Issue by raidnet-ms** [esphome/issues#2245]
### The problem

Hello,
I'm having issues to make this sensor work. Just want to be sure if wemos shield eith SHT-30 is supposed to work with sht3xd platform.

### What is version of ESPHome has the issue?

V1.19.4

### What type of installation are you using?

Home Assistant Add-on

### What platform are you using?

ESP8266

### Board

D1mini

### Component causing the issue

Sht3xd sensor

### Example YAML snippet

```yaml
i2c:
  sda: D2
  scl: D1
# frequency: 1000 Hz
  scan: false

sensor:
  - platform: sht3xd
    address: 0x44 #also tryied 0x45
    Temperature: ...
    Humidity ...


I tryeid reducing i2c frequency to 1000 but i always get nack value for the sensor.
Setting scan to true will cause device timeout.
```

### Anything in the logs that might be useful for us?

```
INFO Waiting for result... INFO OTA successful INFO Successfully uploaded program. INFO Starting log output from termostato-studio.local using esphome API INFO Connecting to termostat.local:6053 (192.168.X.XX) INFO Successfully connected to termostat.local [13:50:48][I][app:105]: ESPHome version 1.20.0 compiled on Jul 23 2021, 13:47:28 [13:50:48][C][status_led:019]: Status LED: [13:50:48][C][status_led:020]:   Pin: GPIO2 (Mode: OUTPUT) [13:50:48][C][wifi:499]:

...Wifi log...

[13:50:48][C][i2c:028]: I2C Bus: [13:50:48][C][i2c:029]:   SDA Pin: GPIO4 [13:50:48][C][i2c:030]:   SCL Pin: GPIO5 [13:50:48][C][i2c:031]:   Frequency: 50000 Hz [13:50:48][I][i2c:033]: Scanning i2c bus for active devices... [13:50:48][I][i2c:040]: Found i2c device at address 0x45 [13:50:48][C][switch.gpio:048]: GPIO Switch 'heater' [13:50:48][C][switch.gpio:049]:   Pin: GPIO12 (Mode: OUTPUT) [13:50:48][C][switch.gpio:071]:   Restore Mode: Restore (Defaults to OFF) [13:50:48][C][logger:189]: Logger: [13:50:48][C][logger:190]:   Level: DEBUG [13:50:48][C][logger:191]:   Log Baud Rate: 115200 [13:50:48][C][logger:192]:   Hardware UART: UART0 [13:50:48][C][sht3xd:034]: SHT3xD: [13:50:48][C][sht3xd:035]:   Address: 0x45 [13:50:48][E][sht3xd:037]: Communication with SHT3xD failed! [13:50:48][C][sht3xd:039]:   Update Interval: 60.0s [13:50:48][C][sht3xd:041]:   Temperature 'Termostat temp' [13:50:48][C][sht3xd:041]:     Device Class: 'temperature' [13:50:48][C][sht3xd:041]:     State Class: 'measurement' [13:50:48][C][sht3xd:041]:     Unit of Measurement: '°C' [13:50:48][C][sht3xd:041]:     Accuracy Decimals: 1 [13:50:48][C][sht3xd:042]:   Humidity 'Termostat hue' [13:50:48][C][sht3xd:042]:     Device Class: 'humidity' [13:50:48][C][sht3xd:042]:     State Class: 'measurement' [13:50:48][C][sht3xd:042]:     Unit of Measurement: '%' [13:50:48][C][sht3xd:042]:     Accuracy Decimals: 1 [13:50:48][C][thermostat.climate:538]: Thermostat 'Termostat' [13:50:48][C][thermostat.climate:543]:   Default Target Temperature Low: 20.0°C [13:50:48][C][thermostat.climate:551]:   Hysteresis: 0.5°C [13:50:48][C][thermostat.climate:552]:   Supports AUTO: NO [13:50:48][C][thermostat.climate:553]:   Supports HEAT/COOL: NO [13:50:48][C][thermostat.climate:554]:   Supports COOL: NO [13:50:48][C][thermostat.climate:555]:   Supports DRY: NO [13:50:48][C][thermostat.climate:556]:   Supports FAN_ONLY: NO [13:50:48][C][thermostat.climate:557]:   Supports HEAT: YES [13:50:48][C][thermostat.climate:558]:   Supports FAN MODE ON: NO [13:50:48][C][thermostat.climate:559]:   Supports FAN MODE OFF: NO [13:50:48][C][thermostat.climate:560]:   Supports FAN MODE AUTO: NO [13:50:48][C][thermostat.climate:561]:   Supports FAN MODE LOW: NO [13:50:48][C][thermostat.climate:562]:   Supports FAN MODE MEDIUM: NO [13:50:48][C][thermostat.climate:563]:   Supports FAN MODE HIGH: NO [13:50:48][C][thermostat.climate:564]:   Supports FAN MODE MIDDLE: NO [13:50:48][C][thermostat.climate:565]:   Supports FAN MODE FOCUS: NO [13:50:48][C][thermostat.climate:566]:   Supports FAN MODE DIFFUSE: NO [13:50:48][C][thermostat.climate:567]:   Supports SWING MODE BOTH: NO [13:50:48][C][thermostat.climate:568]:   Supports SWING MODE OFF: NO [13:50:48][C][thermostat.climate:569]:   Supports SWING MODE HORIZONTAL: NO [13:50:48][C][thermostat.climate:570]:   Supports SWING MODE VERTICAL: NO [13:50:48][C][thermostat.climate:571]:   Supports TWO SET POINTS: NO [13:50:48][C][thermostat.climate:572]:   Supports AWAY mode: NO [13:50:48][C][captive_portal:148]: Captive Portal: [13:50:48][C][ota:029]: Over-The-Air Updates: [13:50:48][C][ota:030]:   Address:   Using Password. [13:50:48][C][api:095]: API Server: [13:50:48][C][api:096]:   

```

**Comment by intermittech** [comment]
Try and enable the I2C scan and see if it at least sees something on the bus? Also maybe try to use the GPIO numbers, not the "D" numbers stuff.

**Comment by raidnet-ms** [comment]
Hi, @intermittech if I enable scanning the device get stuck on "scanning". I can see that trough the serial log because in wifi won't connect if scan is enabled.

Already tried with gpio4 and 5.

So you can confirm that sht30 v1.0 shield should work platform sht3xd?

**Comment by agners** [comment]
I do have SHT3x sensors working with ESPHome v1.19. It should also work in that particular combination, also reported by others:
https://community.home-assistant.io/t/esphome-wemos-d1-mini-pro-v2-sht30-working-hp303b-pressure-shield-not-working/166711/8

Maybe v1.0 has no pull-ups or something? It seems that v2.1 is current?
https://www.wemos.cc/en/latest/d1_mini_shield/sht30.html

**Comment by raidnet-ms** [comment]
I had a doubt related to the relay shield that was connected upon the sensor sharing the D1 pin (of course I had it disabled).
Btw I soldered pins to the sensor now so I could connect directly on the d1mini with no other interference but I still get  [E][sht3xd:037]: Communication with sht3xd failed!

The i2c scan now is not getting stuck anymore but it does not find any device.
I also tryed with 0x44 and 0x45 address with no success.
Maybe the v1.0 of sjt30 shield is not supported?

**Comment by raidnet-ms** [comment]
> I do have SHT3x sensors working with ESPHome v1.19. It should also work in that particular combination, also reported by others:
> h



> I do have SHT3x sensors working with ESPHome v1.19. It should also work in that particular combination, also reported by others:
> https://community.home-assistant.io/t/esphome-wemos-d1-mini-pro-v2-sht30-working-hp303b-pressure-shield-not-working/166711/8
> 
Yup I already trued this way.

**Comment by raidnet-ms** [comment]
Solved with esphome 1.20 🤷
